### ISSUE TITLE
Train API endpoint should be asynchronous

### ISSUE THREAD

**Issue by ciccio86** [OpenReqEU/requirements-classifier#3]
The endpoint `POST /upc/classifier-component/train` is time consuming and is currently synchronous.

It should be possible to call the API in an asynchronous way, i.e. it should return immediately notifying that the training has been taken in charge.
Moreover an endpoint indicating the status of a model should be provided.

## Example

```
GET /upc/classifier-component/check-status?company=test&property=test
```

and should return the status of the training for the pertaining model for example:

```json
{
    "company": "test",
    "property": "test",
    "status": "done"
}
```

Possible statuses:

- done (training successfully completed)
- pending (model currently in training)
- error (training failed with error)
- non-existent (the model does not exist)

**Comment by quim-motger** [comment]
Dear @ciccio86 ,

we are currently developing an asynchronous communication pattern in some other time-consuming methods of other OpenReq components such as the similarity-detection service, so we can use the same structure that is being developed there in orther to maintain consistency.

We will consider this as an enhancement task for this service and we will add it to one of the nearest future versions. Once a release with this new feature is published, I will report it here and I will close the issue.

Thank you for your considerations.

Best regards,

Quim Motger

**Comment by dfucci** [comment]
@ciccio86 can you open a pull request to integrated the async solution you showed me and close this issue?

**Comment by ciccio86** [comment]
@dfucci the solution in the demo I showed you is implemented in the application using the requirements-classifier service.
The communication between the application and the service itself is indeed synchronous.

**Comment by dfucci** [comment]
> The communication between the application and the service itself is indeed synchronous.

@ciccio86 Can you open a pull request to integrate this feature?

**Comment by dfucci** [comment]
Closed in 78c43d437e46ef139607b60206ac1fbca245c981  
See release [v3.0](https://github.com/OpenReqEU/requirements-classifier/tree/3.0)

**Comment by quim-motger** [comment]
Both binary and multilabel train endpoints are now asynchronous. In order to keep the communication as simple as possible, we have implemented an observer pattern, where an URL is required as the service listener where the response will be sent once the training has finished. This way it is not necessary to continuously poll the service waiting for the response.

See release v3.0 and the swagger documentation for major details. Notice this version is not compatible with v2.0, which interface was intended to be synchronous.
